src/antsibull_changelog/config.py

```diff
@@ -362,6 +362,177 @@ def get_flatmap(self) -> bool | None:
 ]
 
 
+class ChangelogRenderConfig(p.BaseModel):
+    """
+    Configuration for a changelog output.
+    """
+
+    model_config = p.ConfigDict(frozen=True, extra="allow", validate_default=True)
+
+    title_version_depth: int = 0
+
+    global_toc: bool = True
+    global_toc_depth: t.Optional[int] = None
```

Comment:
You can do this and then remove the validator:

```suggestion
    global_toc_depth: t.Optional[p.PositiveInt] = None
```

Reply:
Thanks, that simplifies things a bit :) (Changed in d96dfa30d256d92f16e7e6a463bd595fb5b8003b, I also used `p.NonNegativeInt` to get rid of the other validator as well.)

(And sorry for replying so late, I was busy with work-related stuff and data tagging...)